server/src/main/java/org/apache/druid/server/initialization/ServerConfig.java

```diff
@@ -145,6 +150,13 @@ public ServerConfig()
   @NotNull
   private List<String> allowedHttpMethods = ImmutableList.of();
 
+  @JsonProperty
+  @NotNull
+  private List<Pattern> responseWhitelistRegex = ImmutableList.of();
```

Comment:
this config applies to the http servers, not queries specifically, so I think if it is going to live here it might need a name that indicates it only applies to queries, unless you imagine that at some point this setting will apply to all HTTP error responses?

Alternatively, if it lived on a more query centric config object this wouldn't be necessary, though i'm not quite sure off the top of my head what is most appropriate.

Reply:
I imagine the setting for filterResponse will apply to all HTTP error responses. Currently it does apply to API outside of query APIs if the error is thrown in the Jetty filter layer. For example, if you access non-query API with invalid auth or an invalid API endpoint.

The setting for responseWhitelistRegex...I am not sure yet. I think it can apply to all HTTP error responses. Another idea I had was instead of using a whitelist, we can use a blacklist regex. Then it would be simpler to blacklist with regex like `org\.apache\..*` and `java\..*` to filter out stack traces.  This would then be easy to apply to all HTTP error repsonses.